sbin/common/common.sh

```diff
@@ -173,7 +173,30 @@ function setBootJdk() {
     fi
 
     echo "Guessing JDK_BOOT_DIR: ${BUILD_CONFIG[JDK_BOOT_DIR]}"
-    echo "If this is incorrect explicitly configure JDK_BOOT_DIR"
+    echo "=BOOT JDK VERSION OUTPUT="
+    ${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -version
+    echo "=/BOOT JDK VERSION OUTPUT="
+
+    # Calculate version number of boot jdk
+    export BOOT_JDK_VERSION_NUMBER=$(${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -XshowSettings:properties -version 2>&1 | awk '/java.specification.version/{gsub(/1\./,"",$3);print $3}')
+    if [ -z "$BOOT_JDK_VERSION_NUMBER" ]; then
+      echo "[ERROR] The BOOT_JDK_VERSION_NUMBER was not found. Likelihood is that the boot jdk settings properties don't contain a java.specification.version..."
+      ${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -XshowSettings:properties -version 2>&1
+      exit 2
+    fi
+
+    # If bootjdk isn't the same version and isn't a n-1 version, crash out with informational message
+    export REQUIRED_BOOT_JDK=$((${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}-1))
+    # Unless it's jdk8 in which case, a jdk8 boot dir is fine
+    if [ ${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]} -eq 8 ]; then
+      REQUIRED_BOOT_JDK=${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}
```

Comment:
If I'm reading this correctly, isn't this going to fail if we're using a JDK7 to bootstrap JDK8, which is the normal case? I'm not sure what situation this is designed to trap that isn't already covered in the following `if` block

Reply:
Yeah I may have gone a little overboard there trying to catch every eventuality. The bottom `if` does indeed do the same as this `if` so this one can be removed